From 1b22b45c3a3ff52ea11e3389b9c7c57c1daf9798 Mon Sep 17 00:00:00 2001
From: root <root@vpn01.ffv>
Date: Mon, 5 Dec 2016 19:12:10 +0100
Subject: [PATCH] Add unmodified src/dst mac in each graph link

Only the primary mac of the device will be stored in the final
graph. But the ffv-meshviewer-filter requires the per-link mac
address for destination and source to calculate the type of link.
---
 lib/graph.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/graph.py b/lib/graph.py
index db1259e..9a7fae4 100644
--- a/lib/graph.py
+++ b/lib/graph.py
@@ -18,7 +18,7 @@ def import_vis_data(graph, nodes, vis_data):
 
     edges = filter(lambda d: 'neighbor' in d, vis_data)
     graph.add_edges_from(map(lambda d: (d['router'], d['neighbor'],
-                                        dict(tq=float(d['label']))), edges))
+                                        dict(tq=float(d['label']),src=d['router'],dst=d['neighbor'])), edges))
 
 
 def mark_vpn(graph, vpn_macs):
-- 
2.1.4

